source/SpinalHDL/Data types/Int.rst

```diff
@@ -338,52 +345,136 @@ To cast a ``Bool``, a ``Bits``, or an ``SInt`` into a ``UInt``, you can use ``U(
    // Cast a Bits to SInt
    val mySInt = S(myBits)
 
+   // UInt to SInt conversion
+   val UInt_30 = U(30, 8 bit)
+
+   val SInt_30 = UInt_30.intoSInt
+   assert(SInt_30 === S(30, 9 bit))
+
+   mySInt := UInt_30.twoComplement(booleanDoInvert)
+       // if booleanDoInvert is True then we get S(-30, 9 bit)
+       // otherwise we get S(30, 9 bit)
+
+   // absolute values
+   val SInt_n_4 = S(-3, 3 bit)
+   val abs_en = SInt_n_3.abs(booleanDoAbs)
+       // if booleanDoAbs is True we get U(3, 3 bit)
+       // otherwise we get U"3'b101" or U(5, 3 bit) (raw bit pattern of -3)
+
+   val SInt_n_128 = S(-128, 8 bit)
+   val abs = SInt_n_128.abs
+   assert(abs === U(128, 8 bit))
+   val sym_abs = SInt_n_128.absWithSym
+   assert(sym_abs === U(127, 7 bit))
+
 Bit extraction
-~~~~~~~~~~~~~~
+^^^^^^^^^^^^^^
+
+All of the bit extraction operations can be used to read a bit / group of bits. Like in other HDLs
+the extraction operators can also be used to assign a part of a ``UInt`` / ``SInt`` .
 
 .. list-table::
    :header-rows: 1
-   :widths: 2 6 2
+   :widths: 2 4 2
 
    * - Operator
      - Description
      - Return
-   * - x(y)
-     - Readbit, y : Int/UInt
+   * - x(y: Int)
+     - Static bit access of y-th bit
+     - Bool
+   * - x(x: UInt)
+     - Variable bit access of y-th bit
+     - Bool
+   * - x(offset: Int, width bits)
+     - Fixed part select of fixed width, offset is LSB index
+     - Bits(width bits)
+   * - x(offset: UInt, width bits)
+     - Variable part-select of fixed width, offset is LSB index
+     - Bits(width bits)
+   * - x(range: Range)
+     - Access a :ref:`range <range>` of bits. Ex : myBits(4 downto 2)
+     - Bits(range.size bits)
+   * - x.subdivideIn(y slices, [strict: Boolean])
+     - Subdivide x into y slices, y: Int
+     - Vec(Bits(...), y)
+   * - x.subdivideIn(y bits, [strict: Boolean])
+     - Subdivide x in multiple slices of y bits, y: Int
+     - Vec(Bits(y bit), ...)
+   * - x.msb
+     - Access most significant bit of x (highest index)
```

Comment:
Is it worth pointing out how this might vary or not-vary depending upon if x is signed (SInt) or unsigned (UInt/Bits) ?

That is not obvious from reading the docs, especially when `mySInt.sign` is just below.

Reply:
Is that identical to `x.sign` in the case of SInt types ?  or is that the highest MSB of the range `x(x.getWidth - 1)` ?
That is the clarification to make.

Reply:
added that as a note